ui/src/app/stores/workflows/workflows.effects.ts

```diff
@@ -247,50 +248,64 @@ export class WorkflowsEffects {
     ofType(WorkflowActions.CREATE_WORKFLOW),
     withLatestFrom(this.store.select(selectWorkflowState)),
     switchMap(([action, state]: [WorkflowActions.CreateWorkflow, fromWorkflows.State]) => {
-      const workflowCreateRequest = new WorkflowRequestModel(
-        state.workflowAction.workflowFormData.details,
-        state.workflowAction.workflowFormData.sensor,
-        state.workflowAction.workflowFormData.jobs,
-      ).getCreateWorkflowRequestObject();
-
-      return this.workflowService.createWorkflow(workflowCreateRequest).pipe(
-        mergeMap((result: WorkflowJoinedModel) => {
-          const workflow: WorkflowModel = WorkflowModelFactory.create(
-            result.name,
-            result.isActive,
-            result.project,
-            result.created,
-            result.updated,
-            result.id,
-          );
-          this.toastrService.success(texts.CREATE_WORKFLOW_SUCCESS_NOTIFICATION);
-          this.router.navigateByUrl(absoluteRoutes.SHOW_WORKFLOW + '/' + workflow.id);
+      return this.workflowService.getJobTemplates().pipe(
```

Comment:
This is quite a large effort just to get 'Spark' or 'Shell' :) But ok, since creation/editing of workflows is rare, I'm ok with this.

Reply:
Yes, but I didn't have other option :/